tools/contractdeployer/contract_deployer.go

```diff
@@ -119,18 +120,18 @@ func (cd *contractDeployer) run(isL1Deployment bool) error {
 
 	contractAddr, err := signAndSendTxWithReceipt(cd.wallet, cd.client, &deployContractTx)
 	if err != nil {
-		return err
+		return "", err
 	}
 	if contractAddr == nil {
-		return fmt.Errorf("transaction was successful but could not retrieve address for deployed contract")
+		return "", fmt.Errorf("transaction was successful but could not retrieve address for deployed contract")
 	}
 
 	// print the contract address, to be read if necessary by the caller (important: this must be the last message output by the script)
 	fmt.Print(contractAddr.Hex())
```

Comment:
I think this should move to the main() now that the deployer returns the address - makes it easier to guarantee it's the last output line.

Reply:
Yeah, good idea.